src/main/java/edu/project2/tripplanner/controller/UserController.java

```diff
@@ -0,0 +1,69 @@
+package edu.project2.tripplanner.controller;
+
+import edu.project2.tripplanner.model.User;
+import edu.project2.tripplanner.service.IUserService;
+import lombok.AllArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.*;
+
+@AllArgsConstructor
+@RestController
+public class UserController {
+
+    IUserService iUserService;
+
+    @ResponseBody
+    @GetMapping("/user")
+    public List<User> findAllUser() {
+
+        return iUserService.findAllUsers();
+    }
+
+
+    @ResponseBody
+    @PostMapping("/user")
+    public ResponseEntity<User> saveUser(@RequestBody User user) {
+        iUserService.saveUser(user);
+        return ResponseEntity
+                .status(HttpStatus.CREATED)
+                .build();
+    }
+
+
+    @ResponseBody
+    @DeleteMapping("/user/{username}")
+    public void deleteUser(@PathVariable("username") String username) {
+        iUserService.deleteByUsername(username);
+    }
+
+
+    @ResponseBody
+    @PatchMapping("/user")
+    public User updateUser(@RequestBody User user) {
+        return iUserService.updateUser(user);
+    }
+
+
+    @ResponseBody
+    @GetMapping("/user/id/{id}")
+    public Optional<User> getUserById(@PathVariable("id") Long id) {
+        return iUserService.getUserById(id);
+    }
+
+    @ResponseBody
+    @GetMapping("/user/{username}")
+    public Optional<User> findByUsername(@PathVariable("username") String username) {
+        return iUserService.findByUsername(username);
+    }
+
+    @ResponseBody
+    @GetMapping("/user/exist/{username}")
```

Comment:
what is the value of this API?
You should always return a structure. Even if it is one field there.
Create DTO if you want to use this API. But actually, I didn't get why do you need this one